sdks/go/pkg/beam/io/xlang/bigqueryio/bigquery.go

```diff
@@ -0,0 +1,222 @@
+// Required fields are value types. Repeated fields are represented as slices in Go (and ARRAYS
+// in SQL, as in the table above).
+//
+// Example of BigQuery fields with modes:
+//   field1 *int64   // Nullable INT64
+//   field2 int64    // Required INT64
+//   field3 []int64  // Repeated INT64
+//
+// Note On Documentation
+//
+// This cross-language implementation relies on the behavior of external SDKs. In order to keep
+// documentation up-to-date and consistent, BigQuery functionality will not be described in detail
+// in this package. Instead, references to relevant documentation in other SDKs is included where
+// relevant.
+package bigqueryio
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/typex"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/io/xlang/schemaio"
+)
+
+type createDisposition string
+
+const (
+	// CreateNever specifies that new tables should not be created when writing to BigQuery.
+	CreateNever createDisposition = "Never"
+
+	// CreateIfNeeded specifies that tables should be created when writing to BigQuery, if needed.
+	CreateIfNeeded createDisposition = "IfNeeded"
+
+	readURN  = "beam:transform:org.apache.beam:schemaio_bigquery_read:v1"
+	writeURN = "beam:transform:org.apache.beam:schemaio_bigquery_write:v1"
+)
+
+// bigQueryConfig is a struct meant to match the Schema IO config for Java's BigQuery IO. This is
+// used for both reads and writes, and is meant to match the schema defined in the Java SDK method
+// org.apache.beam.sdk.io.gcp.bigquery.BigQuerySchemaIOProvider.configurationSchema().
+type bigQueryConfig struct {
+	Table             *string `beam:"table"`
+	Query             *string `beam:"query"`
+	QueryLocation     *string `beam:"queryLocation"`
+	CreateDisposition *string `beam:"createDisposition"`
+}
+
+// Read is a cross-language PTRansform which reads from a BigQuery table and returns a PCollection
+// of the given type, which should correspond to the Schema type generated by reading from the
+// table.
+//
+// Read requires the address for an expansion service for BigQuery Read transforms and a
+// reflect.Type description of the struct to read from BigQuery. Additionally, either one Table
+// or one Query must be provided via readOptions to define a destination to read from.
+//
+// Read accepts additional parameters as readOptions. All optional parameters are predefined in this
+// package as functions that return readOption. To set an additional parameter, call the function
+// within Read's function signature.
+//
+// Example:
+//   expansionAddr := "localhost:1234"
+//   table := "project_id:dataset_id.table_id"
+//   outType := reflect.TypeOf((*Foo)(nil)).Elem()
+//   pcol := bigqueryio.Read(s, expansionAddr, outType, bigqueryio.FromTable(table))
+func Read(s beam.Scope, addr string, elmT reflect.Type, opts ...readOption) beam.PCollection {
```

Comment:
Nit, have the expansion address happen as an optional parameter, so we can do the auto expansion start by default later. This means we have the documentation show the use of the option, and until we have the auto-start in, we have the code require it. Just not by a compile time restriction.

Reply:
Done. Also added expansion service auto-startup.